ui/ui-components/utils/profileHandler.ts

```diff
@@ -1,3 +1,4 @@
+import { Models } from "./apiData";
 import { EventDispatcher } from "./eventDispatcher";
 
-export class ProfileHandler extends EventDispatcher {}
+export class ProfileHandler extends EventDispatcher<Models.ProfileType> {}
```

Comment:
This is pretty neat - we can just reference the type returned from the API/Model's serializer.